File: src/templates/skills/codi-brand/scripts/python/brand_tokens.py
"""
brand_tokens.py — Python adapter for Codi brand tokens.
Reads brand_tokens.json and exposes typed constants for python-pptx use.
"""

import json
import os

_DIR = os.path.dirname(__file__)
_JSON = os.path.join(_DIR, "..", "brand_tokens.json")

with open(_JSON, "r", encoding="utf-8") as _f:
    _data = json.load(_f)

COLORS: dict[str, str] = _data["colors"]
FONTS: dict[str, str] = _data["fonts"]
LAYOUT: dict[str, str] = _data["layout"]
ASSETS: dict[str, str] = _data["assets"]
VOICE: dict = _data["voice"]


def hex_color(key: str) -> str:
    """Return 6-char hex string without # (for pptx RGBColor)."""
    color = COLORS.get(key, "")
    if not color:
        raise ValueError(
            f"Unknown Codi color token: '{key}'. Available: {list(COLORS)}"
        )
    return color.lstrip("#")


def rgb(key: str) -> tuple[int, int, int]:
    """Return (r, g, b) tuple for a color token."""
    h = hex_color(key)
    return int(h[0:2], 16), int(h[2:4], 16), int(h[4:6], 16)

File: src/templates/skills/codi-brand/scripts/ts/brand_tokens.ts
/**
 * brand_tokens.ts — TypeScript adapter for Codi brand tokens.
 * Reads brand_tokens.json and re-exports typed constants for pptxgenjs use.
 */
import { readFileSync } from "node:fs";
import { join, dirname } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const raw = JSON.parse(
  readFileSync(join(__dirname, "..", "brand_tokens.json"), "utf-8")
) as Record<string, unknown>;

export const COLORS = raw["colors"] as Record<string, string>;
export const FONTS  = raw["fonts"]  as Record<string, string>;
export const LAYOUT = raw["layout"] as Record<string, string>;
export const ASSETS = raw["assets"] as Record<string, string>;
export const VOICE  = raw["voice"]  as { phrases_use: string[]; phrases_avoid: string[] };

/** Returns hex color without # prefix (pptxgenjs expects bare hex strings). */
export function hex(key: string): string {
  const color = COLORS[key];
  if (!color) throw new Error(`Unknown Codi color token: "${key}". Available: ${Object.keys(COLORS).join(", ")}`);
  return color.replace("#", "");
}

export const GOOGLE_FONTS_URL =
  "https://fonts.googleapis.com/css2" +
  "?family=Outfit:wght@400;500;700;800" +
  "&family=Geist+Mono:wght@400;500" +
  "&display=swap";
